FAQ.md

```diff
@@ -0,0 +1,104 @@
+
+## I created a Machine but it never joined the cluster
+This can be a variety of things.
+
+First, determine if the machine object’s status has information regarding networking and instance ID’s.  If this information is present, it means the cloud provider accepted the request to create the machine and the machine was created successfully.
+
+If successful, check for any pending CSR requests.  If there are not any pending CSR requests, then the problem is not related to the machine-api.  Possible causes include misconfigured MCO machine config, cloud or networking problems, or a machine that has the wrong user-provided configuration (eg, you set the wrong subnet or image ID).
+
+If the machine takes a very unusual amount of time to start after machine creation, the CSR created for the associated node to join the cluster will not be automatically approved.  This can be due to cloud capacity constraints, cloud api problems, temporary network conditions, or other circumstances.  You can either approve the CSRs (after approving one, another will be created for the same node), or delete the machine and create a new one (the machineset will create a new machine for you if this machine is part of a machineset).
+
+If the machine’s status does not have any instance ID or networking information associated with it, most likely the instance has failed to be created.  This can be due to misconfiguration, inadequate or invalid cloud credentials, cloud API quota exhaustion, or other cloud API problems such as an outage or temporary network condition.  You will need to inspect the machine-controllers logs for more definitive information.
+
+## What happens when I delete a Machine?
+This section assumes you’ve deleted a machine from the API, rather than deleting a VM or Instance in the cloud provider.
+
+When a machine is marked deleted, the associated node is immediately cordoned and drained.  Draining utilizes the eviction API, so PodDisruptionBudgets will be respected.  If a node cannot be successfully drained due to a PDB, the operation will retry indefinitely.
+
+After an associated node is drained, the associated instance or VM is deleted from the cloud provider.  After the instance is deleted in the cloud provider, the associated node object is deleted from the API.
+
+Finally, the finalizer is removed from the machine object and the machine object is removed from the API.
+
+## I want to skip draining when I delete a Machine
+This is not recommended for most cases, especially Master Machines.  Properly draining machines will respect PodDisruptionBudgets and prevent the cluster and workloads from going into an unhealthy state.
+
+You can optionally set an **annotation** **"machine.openshift.io/exclude-node-draining"** on each machine object you wish for draining to be skipped.  This can be applied or removed at any time.
```

Comment:
Might be worth clarifying the value of the annotation rather than just the name? (I assume it doesn't matter?)

Reply:
Value doesn't matter.  I can clarify.